lib/proxy.js

```diff
@@ -94,9 +94,9 @@ function buildStream (host, options, secure) {
 ////////////////////////////////////////////////////////////
 
 function buildResultingMiddleware (host, options, stream) {
-  console.log(stream)
   return function proxyMiddleware (req, res, next) {
 
+    var _stream = _.clone(stream, true);
```

Comment:
I think this might be more the desired behavior:
`96 |  function buildResultingMiddleware (host, options, sharedStream) {`
`99 | var stream = _.clone(sharedStream)`